velox/type/tests/StringViewTest.cpp

```diff
@@ -156,3 +159,129 @@ TEST(StringView, negativeSizes) {
+    auto item = string;
+    for (auto repeat = 0; repeat < numRepeats; ++repeat) {
+      string += item;
+    }
+    string.resize(std::min<int32_t>((i >= 50 ? 2 : 0) + string.size(), i % 50));
+    data[i].resize(string.size());
+    if (!string.empty()) {
+      memcpy(data[i].data(), string.data(), string.size());
+    }
+    stringViews[i] = StringView(data[i].data(), data[i].size());
+  }
+  raw_vector<int32_t> indices(kSize);
+  for (auto i = 0; i < kSize; ++i) {
+    indices[i] = 999 - i;
+  }
+
+  uint64_t simdUsec = 0;
+  uint64_t loopUsec = 0;
+  uint64_t simdIndicesUsec = 0;
+  uint64_t loopIndicesUsec = 0;
+  for (auto counter = 0; counter < 10; ++counter) {
+    {
+      MicrosecondTimer t(&simdUsec);
+      EXPECT_EQ(
+          -1,
+          StringView::linearSearch(
+              stringViews[11], stringViews.data(), nullptr, 10));
+      for (auto i = 0; i < kSize; ++i) {
+        auto testIndex = (i * 1) % kSize;
+        auto index = StringView::linearSearch(
+            stringViews[testIndex],
+            stringViews.data(),
+            nullptr,
+            stringViews.size());
+        EXPECT_TRUE(stringViews[testIndex] == stringViews[index]);
+      }
+    }
+    {
+      MicrosecondTimer t(&loopUsec);
+      EXPECT_EQ(
+          -1,
+          linearSearchSimple(stringViews[11], stringViews.data(), nullptr, 10));
+      for (auto i = 0; i < kSize; ++i) {
+        auto testIndex = (i * 1) % kSize;
+        auto index = linearSearchSimple(
+            stringViews[testIndex],
+            stringViews.data(),
+            nullptr,
+            stringViews.size());
+        EXPECT_TRUE(stringViews[testIndex] == stringViews[index]);
+      }
+    }
+
+    {
+      MicrosecondTimer t(&simdIndicesUsec);
+      EXPECT_EQ(
+          -1,
+          StringView::linearSearch(
+              stringViews[indices[11]],
+              stringViews.data(),
+              indices.data(),
+              10));
+      for (auto i = 0; i < kSize; ++i) {
+        auto testIndex = (i * 1) % kSize;
+        auto index = StringView::linearSearch(
+            stringViews[testIndex],
+            stringViews.data(),
+            indices.data(),
+            stringViews.size());
+        EXPECT_TRUE(stringViews[testIndex] == stringViews[indices[index]]);
+      }
+    }
+    {
+      MicrosecondTimer t(&loopIndicesUsec);
+      EXPECT_EQ(
+          -1,
+          linearSearchSimple(
+              stringViews[indices[11]],
+              stringViews.data(),
+              indices.data(),
+              10));
+      for (auto i = 0; i < kSize; ++i) {
+        auto testIndex = (i * 1) % kSize;
+        auto index = linearSearchSimple(
+            stringViews[testIndex],
+            stringViews.data(),
+            indices.data(),
+            stringViews.size());
+        EXPECT_TRUE(stringViews[testIndex] == stringViews[indices[index]]);
+      }
+    }
+  }
+  LOG(INFO) << "StringView search: SIMD: " << simdUsec << " / "
```

Comment:
@oerling @Yuhta Why are we logging execution time here? Shouldn't this be written as a benchmark?

Reply:
Yes it serves as a benchmark as well.  It would be troublesome if it was running longer but it is small in this case (about 2s in debug mode) so does not hurt to put it with unit test.

Reply:
@Yuhta Not having it written as a benchmark prevents us from running it as a benchmark adhoc and on PRs. It would be nice to convert this into a proper benchmark.

Reply:
Crete an issue to track this: #7020
I will keep an eye to separate benchmark and test in future changes.

Reply:
@Yuhta Thanks.